The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10028538514/job/27715579034) [(plain)](https://github.com/rust-lang/rust/commit/36c4e00959090b2de0c70414819f7a593352e61c/checks/27715579034/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
  CACHE_DOMAIN: ci-caches.rust-lang.org
  PR_CI_JOB: 1
  IMAGE: mingw-check-tidy
##[endgroup]
fatal: unknown commit 53fd98ca776cb875bc9e5514f56b52eb74f9e7a9
All commits in `HEAD` are present in `master`
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.448   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.466 Collecting boolean-py==4.0
#13 0.473   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.491 Collecting chardet==5.1.0
---
#13 3.966 Building wheels for collected packages: reuse
#13 3.967   Building wheel for reuse (pyproject.toml): started
#13 4.307   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.307   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.308   Stored in directory: /tmp/pip-ephem-wheel-cache-utauouxt/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.310 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.336   Attempting uninstall: setuptools
#13 4.337     Found existing installation: setuptools 59.6.0
#13 4.338     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.657   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.887      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 25.0 MB/s eta 0:00:00
#13 5.948 Collecting filelock<4,>=3.12.2
#13 5.955   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.989 Collecting platformdirs<5,>=3.9.1
#13 5.996   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 6.019 Collecting distlib<1,>=0.3.7
#13 6.026   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 6.039      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 41.2 MB/s eta 0:00:00
#13 6.126 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 6.314 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 6.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      188352 kB
DirectMap2M:     6103040 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
thread 'main' panicked at src/lib.rs:1543:38:
thread 'main' panicked at src/lib.rs:1543:38:
called `Result::unwrap()` on an `Err` value: ParseIntError { kind: Empty }
Build completed unsuccessfully in 0:00:00
  local time: Sun Jul 21 13:08:30 UTC 2024
  network time: Sun, 21 Jul 2024 13:08:30 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -1511,6 +1511,12 @@ Executed at: {executed_at}"#,
     }
 
     fn beta_prerelease_version(&self) -> u32 {
+        if self.config.dry_run() {
+            // Dry run doesn't actually execute a git command, which will panic when trying to parse
+            // the outputs.
+            return 0;
+        }
+
         fn extract_beta_rev_from_file<P: AsRef<Path>>(version_file: P) -> Option<String> {
             let version = fs::read_to_string(version_file).ok()?;
 

